### PR TITLE
vscode: fix linking errors

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Wasn't able to launch it without the following modifications.

Also tested vscodium.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124850` at e2b66794 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>vscode</li>
    <li>vscode-with-extensions</li>
    <li>vscodium</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Does this need a backport?

**Comment by jonringer** [comment]
Yes

**Review by eadwu — APPROVED** [review]
Can't reproduce with `vscode` but it doesn't hurt anyway.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 124850` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>vscode</li>
    <li>vscode-with-extensions</li>
    <li>vscodium</li>
  </ul>
</details>

File: pkgs/applications/editors/vscode/generic.nix [hunk 1]
```diff
@@ -14,7 +15,7 @@
 
 let
   inherit (stdenv.hostPlatform) system;
-  unwrapped = stdenv.mkDerivation {
+  unwrapped = stdenv.mkDerivation rec {
```
Comment (turion) [inline pkgs/applications/editors/vscode/generic.nix:18]: Some people say that `rec` is an antipattern. I'm not sure what the nixpkgs consensus is.

**Comment by turion** [comment]
What happened to the tests? I don't understand the error message.

File: pkgs/applications/editors/vscode/generic.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/applications/editors/vscode/generic.nix:18]: I needed to reference buildInputs. I could deitehr factor it outside the attrset, or just add rec

Comment (Synthetica9) [reply pkgs/applications/editors/vscode/generic.nix:18]: `rec` when you don't need it is bad (or at least I warn for it nix-linter), but I don't think I've heard anyone say that `rec` in general is an antipattern.

**Comment by jonringer** [comment]
this could also just be related to nixUnstable issues... or my nix-daemon locally is causing me grief.

File: pkgs/applications/editors/vscode/generic.nix (hunk 1 above)
Comment (turion) [reply pkgs/applications/editors/vscode/generic.nix:18]: It's been mentioned here: https://nix.dev/anti-patterns/language#rec-expression

Comment (SuperSandro2000) [reply pkgs/applications/editors/vscode/generic.nix:18]: Using it libs and pure functions can lead to confusion but in mkDerivation it is totally normal to use and encouraged.

**Comment by jonringer** [comment]
I'm no longer having this issue.

I think it was just related to a bad version of `nixUnstable`

**Pull request closed without merging**